ui/apps/platform/cypress/helpers/violations.js

```diff
@@ -0,0 +1,128 @@
+import * as api from '../constants/apiEndpoints';
+import { url, selectors } from '../constants/ViolationsPage';
+
+import { visitFromLeftNav } from './nav';
+import { visit } from './visit';
+
+// visit
+
+export function visitViolationsFromLeftNav() {
+    cy.intercept('GET', api.alerts.alertsWithQuery).as('getAlerts');
+    cy.intercept('GET', api.alerts.alertsCountWithQuery).as('getAlertsCount');
+
+    visitFromLeftNav('Violations');
+
+    cy.wait(['@getAlerts', '@getAlertsCount']);
+    cy.get('h1:contains("Violations")');
+}
+
+export function visitViolations() {
+    cy.intercept('GET', api.alerts.alertsWithQuery).as('getAlerts');
+    cy.intercept('GET', api.alerts.alertsCountWithQuery).as('getAlertsCount');
+
+    visit(url);
+
+    cy.wait(['@getAlerts', '@getAlertsCount']);
+    cy.get('h1:contains("Violations")');
+}
+
+export function visitViolationsWithFixture(fixturePath) {
+    cy.fixture(fixturePath).then(({ alerts }) => {
+        const count = alerts.length;
+
+        cy.intercept('GET', api.alerts.alertsWithQuery, {
+            body: { alerts },
+        }).as('getAlerts');
+        cy.intercept('GET', api.alerts.alertsCountWithQuery, {
+            body: { count },
+        }).as('getAlertsCount');
+
+        visit(url);
+
+        cy.wait(['@getAlerts', '@getAlertsCount']);
+        cy.get('h1:contains("Violations")');
+    });
+}
+
+export function visitViolationsWithAlertsForErrorBoundary(alerts) {
+    const count = alerts.length;
+
+    cy.intercept('GET', api.alerts.alertsWithQuery, {
+        body: { alerts },
+    }).as('getAlerts');
+    cy.intercept('GET', api.alerts.alertsCountWithQuery, {
+        body: { count },
+    }).as('getAlertsCount');
+
+    visit(url);
+
+    cy.wait(['@getAlerts', '@getAlertsCount']);
+    // Do not get h1 because goal of this function is to render error boundary.
+}
+
+/*
+ * Assume that current location is violations table with compatible fixture for alerts.
+ */
+export function visitViolationFromTableWithFixture(fixturePath) {
+    cy.fixture(fixturePath).then((alert) => {
+        const { id, policy } = alert;
+        const { name } = policy;
+
+        cy.intercept('GET', `${api.alerts.alerts}/${id}`, {
+            body: alert,
+        }).as('getAlert');
+
+        // Assume table has only one row which has the policy name.
```

Comment:
Is this always true? If someone passed a fixture to this function with an array of alerts would it error out at the destructuring above? If the intent is that this function should always be called with a fixture with a single item, would it be helpful to immediately throw in the resolved promise with a descriptive error message?

Reply:
Aha, the grammar of the comment is ambiguous. I will edit to express the intended meaning.

If `cy.get(…)` matches more than one row, then cypress will throw an error, because by default an error for a test to click multiple elements at once.